pkg/cloud/services/networking/securitygroups.go

```diff
@@ -418,8 +421,9 @@ func (s *Service) reconcileGroupRules(desired, observed infrav1.SecurityGroup) (
 	s.logger.V(4).Info("Deleting rules not needed anymore for group", "name", observed.Name, "amount", len(rulesToDelete))
 	for _, rule := range rulesToDelete {
 		s.logger.V(6).Info("Deleting rule", "ruleID", rule.ID, "groupName", observed.Name)
+		mc := metrics.NewMetricPrometheusContext("security_group_rules", "delete")
```

Comment:
```suggestion
		mc := metrics.NewMetricPrometheusContext("security_group_rule", "delete")
```

Reply:
We should use singular everywhere. Please also take a look if we have plural elsewhere